aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -1117,70 +1114,106 @@ std::tuple<Tensor, Tensor, Tensor> _lu_with_info_cpu(const Tensor& self, bool pi
 
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ triangular_solve ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-template<typename scalar_t>
-static void apply_triangular_solve(Tensor& b, Tensor& A, Tensor& infos, bool upper, bool transpose, bool unitriangular) {
-#ifndef USE_LAPACK
-  AT_ERROR("triangular_solve: LAPACK library not found in compilation");
-#else
-  char uplo = upper ? 'U' : 'L';
-  char trans = transpose ? 'T' : 'N';
-  char diag = unitriangular ? 'U' : 'N';
+DEFINE_DISPATCH(triangular_solve_stub);
 
-  auto A_data = A.data_ptr<scalar_t>();
-  auto b_data = b.data_ptr<scalar_t>();
-  auto A_mat_stride = matrixStride(A);
-  auto b_mat_stride = matrixStride(b);
-  auto batch_size = batchCount(A);
-  auto n = A.size(-2);
-  auto nrhs = b.size(-1);
-  auto lda = std::max<int64_t>(1, n);
-  auto infos_data = infos.data_ptr<int>();
+/*
+Solves the matrix equation 'input' @ 'result' = 'other' for the 'result'.
+The result of the computation is saved in-place in 'result' tensor,
+'clone_input' will be a copy of 'input',
+'infos' is used to store information for possible checks for error,
+'upper' controls the portion of input matrix to consider in computations,
+'transpose' if true then 'input.transpose(-2, -1)' @ 'result' = 'other' is solved,
+'unitriangular' if true then the diagonal elements of 'input' are assumed to be 1
+and the actual diagonal values are not used.
+*/
+static std::tuple<Tensor&, Tensor&> triangular_solve_out_info(
+    Tensor& result,
+    Tensor& clone_input,
+    Tensor& infos,
+    const Tensor& input,
+    const Tensor& other,
+    bool upper, bool transpose, bool unitriangular) {
+  // These internal asserts make explicit the assumptions in the implementation
+  // Error check with the actual error messages are done on the higher level of
+  // the hierarchy of calls
+  TORCH_INTERNAL_ASSERT(input.dim() >= 2);
```

Comment:
If actual error checks are done above, it make sense to make these internal asserts `TORCH_INTERNAL_ASSERT_DEBUG_ONLY`, to avoid overhead of the calls. No need to do it in this PR though